Platform-specific content re-architecture (#1085)
o3de/o3de.org: content/docs/welcome-guide/setup/setup-from-github/_index.md

> ---

Author: In regards to the instructions and screenshots being written for Windows, Steve raised that point too, and my response is that overall, the instructions work well whether you're on Windows or Linux. Unless I'm mistaken, I think the git commands I've shown are interchangeable, right?

Now as for LFS, I'm not sure. @spham-amzn can you comment on @sptramer 's question?

@sptramer , did you encounter the sign in dialog box that's shown in the screenshot, needing your personal access token? I'm not sure that's even needed on Windows any longer...

> ---

Reviewer: There should be no linux variations of the windows based instructions.

The video and written instructions in this section guide you through the following steps:

* Configure credentials for Git LFS.
* Fork and clone the O3DE GitHub repo.
* Build the O3DE engine.
* Register the engine.

{{< youtube id="CIw9UoMMeX8" title="Setting up O3DE from GitHub" >}}

## Prerequisites

The GitHub setup instructions assume that you have:

* [Git client](https://git-scm.com/downloads) installed (1.8.2 or later required, 2.23.4 or later recommended).

## Configure credentials for Git LFS

The O3DE GitHub repo uses the Git Large File Storage (LFS) system for storing large binary files. The following instructions will prepare your PC to authenticate and download these files automatically when you clone, fetch, or pull from the repo.

**To configure for Git LFS**

1. Verify that **Git LFS** is installed.

    ```cmd
    git lfs install
    ```

>     If the output from this command is "Git LFS initialized", then you already have Git LFS installed.

Reviewer 2: What if they don't?

>     If the output from this command is "Git LFS initialized", then you already have Git LFS installed.

Author: ```suggestion
    If the output from this command is "Git LFS initialized", then you already have Git LFS installed. If you didn't have Git LFS installed, this command will install it for you.
```

        # The default is o3de.
        cd o3de
        
        git lfs pull
        ```

1. Add a remote to track the upstream repo. This will enable you to pull updates from the O3DE repo directly into your local clone.

    ```cmd
    git remote add upstream {{< links/o3de-source >}}.git
    ```

    Verify the upstream repository. You should see the URL for the fork as `origin`, and the URL for the original repository as `upstream`.

    ```cmd
    git remote -v
    ```

    Output:

    ```cmd
    origin  https://github.com/<FORK>/o3de.git (fetch)
    origin  https://github.com/<FORK>/o3de.git (push)
    upstream        {{< links/o3de-source >}}.git (fetch)
    upstream        {{< links/o3de-source >}}.git (push)
    ```

1. (Optional) If you don't expect to make changes to these files, ignore this step. Otherwise, update the LFS URL to include your fork. This lets you push changes to files in the LFS. For complete instructions and the **DISTRIBUTION** to use, open the `.lfsconfig` file at the root of the repository.

    ```cmd
    git config lfs.url https://<DISTRIBUTION>.cloudfront.net/api/v1/fork/<FORK> 
    ```

>     You may be prompted to re-authenticate the next time you pull or push.  Remember to use your GitHub personal access token not your GitHub password.

Reviewer 2: ```suggestion
    You might be prompted to re-authenticate the next time that you pull or push.  Remember to use your GitHub personal access token, and not your GitHub password.
```

>     If you wish to revert this change you can run the following command:

Reviewer 2: ```suggestion
    If you wish to revert this change, you can run the following command:
```
